src/core/lib/iomgr/closure.c

```diff
@@ -136,29 +168,45 @@ void grpc_closure_run(grpc_exec_ctx *exec_ctx, grpc_closure *c,
   GPR_TIMER_END("grpc_closure_run", 0);
 }
 
+#ifdef GRPC_CLOSURE_RICH_DEBUG
+void grpc_closure_sched(const char *file, int line, grpc_exec_ctx *exec_ctx,
+                        grpc_closure *c, grpc_error *error) {
+#else
 void grpc_closure_sched(grpc_exec_ctx *exec_ctx, grpc_closure *c,
                         grpc_error *error) {
-  GPR_TIMER_BEGIN("grpc_closure_sched", 0);
+#endif
+  GPR_TIMER_BEGIN("GRPC_CLOSURE_SCHED", 0);
   if (c != NULL) {
-#ifndef NDEBUG
+#ifdef GRPC_CLOSURE_RICH_DEBUG
```

Comment:
Do you think it would be worth it to change all of the custom guards (GRPC_ERROR_REFCOUNT_DEBUG, GRPC_STREAM_REFCOUNT_DEBUG, GRPC_RESOLVER_REFCOUNT_DEBUG, GRPC_LB_POLICY_REFCOUNT_DEBUG) to NDEBUG as well? And maybe additionally turn the into tracers so they can still be turned on and off?

That would solve the problem of the guarded codepaths not being tested on jenkins (to avoid issues like #10691)